### PR TITLE
Analyzer view read only necessary columns

### LINKED ISSUE #62594: Views very slow since upgrading to 24.3
(you don't have to strictly follow this form)

**Describe the situation**
We have several views which only point to a table.
`CREATE VIEW v AS SELECT * FROM myTable`
We don't want our visualization tool (metabase) to directly access tables, in case we need to change something without changing the structure.

Since 24.3 the performance of those views is very bad.
With version 24.2 running queries on the tables directly or the view was basically the same performance.
Now, in 24.3, the views are several times slower.
From what I could tell by checking things like htop during queries, the views now seem to load the entire table into RAM before doing anything.
RAM and CPU spike massively during the query. 

**How to reproduce**
* **Which ClickHouse server version to use** 
24.3
* **Which interface to use, if matters**
Doesn't matter. I could reproduce it with clickhouse-client and HTTP interface. Did not try other, but most likely same problem.
* **Non-default settings, if any**
None
* **`CREATE TABLE` statements for all tables involved**
I was able to reproduce the problem with your sample dataset _UK property prices_
https://clickhouse.com/docs/en/getting-started/example-datasets/uk-price-paid
* **Sample data for all these tables, use [clickhouse-obfuscator]**
Data from the sample.
* **Queries to run that lead to slow performance**
`SELECT max(price) FROM uk_price_paid`
This takes around 15ms on my machine
`CREATE VIEW priceView AS SELECT * FROM uk_price_paid`
`SELECT max(price) FROM priceView`
This takes roughly 100ms on my machine with 24.3.
With 24.2 same performance as direct query on the table.
The bigger the table, the bigger the performance difference between view and table.
Most likely due loading it all first.

**Expected performance**
View and table should have same performance as it has been the case in 24.2 and earlier.

### PR DESCRIPTION
**Pull request by kitaisreal**
### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in an official stable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Read only the necessary columns from VIEW (new analyzer). Closes #62594.

### CODE REVIEW

**Comment by robot-ch-test-poll1** [comment]
*This is an automated comment for commit b8c719f86601780d79de507ef57a0c6e2a8530ad with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/63688/b8c719f86601780d79de507ef57a0c6e2a8530ad/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

File: src/Interpreters/InterpreterSelectQueryAnalyzer.cpp [hunk 1]
```diff
@@ -38,22 +40,46 @@ namespace ErrorCodes
...
-    {
         throw Exception(ErrorCodes::UNSUPPORTED_METHOD,
             "Expected ASTSelectWithUnionQuery or ASTSelectQuery. Actual {}",
             query->formatForErrorMessage());
-    }
+
+    if (column_names.empty())
+        return result_query;
+
+    auto select_query = std::make_shared<ASTSelectQuery>();
```
Comment (novikd) [inline src/Interpreters/InterpreterSelectQueryAnalyzer.cpp:59]:
```suggestion
    /// The initial query the VIEW references to is wrapped here with another SELECT query to allow reading only necessary columns.
    auto select_query = std::make_shared<ASTSelectQuery>();
```

Comment (novikd) [reply src/Interpreters/InterpreterSelectQueryAnalyzer.cpp:59]: Let's add a comment.

**Review by novikd — APPROVED** [review]
LGTM

**Pull request merged**